src/server/rdb_save.cc

```diff
@@ -864,10 +867,15 @@ void RdbSaver::Impl::FillFreqMap(RdbTypeFreqMap* dest) const {
   }
 }
 
-RdbSaver::RdbSaver(::io::Sink* sink, bool single_shard, bool align_writes) {
+RdbSaver::RdbSaver(::io::Sink* sink, SaveMode save_mode, bool align_writes) {
   CHECK_NOTNULL(sink);
 
-  impl_.reset(new Impl(align_writes, single_shard ? 1 : shard_set->size(), sink));
+  int size = 0;
+  if (save_mode == SaveMode::SINGLE_SHARD) size = 1;
```

Comment:
I do not like `if else if` one-liners.
using switch construct will at least cause compiler error if one of the enums is not covered.